src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -1469,6 +1469,8 @@ def create_service_principal_for_rbac(
                     raise
     sp_oid = aad_sp.object_id
 
+    logger.warning(CREDENTIAL_WARNING_MESSAGE)
```

Comment:
Adjust the order of warnings. Show credential warning first.

Before:
```
> az ad sp create-for-rbac --sdk-auth
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590'
  Retrying role assignment creation: 1/36
The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
{
  ...
}
```

After:
```
> az ad sp create-for-rbac --sdk-auth
The output includes credentials that you must protect. Be sure that you do not include these credentials in your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli
In a future release, this command will NOT create a 'Contributor' role assignment by default. If needed, use the --role argument to explicitly create a role assignment.
Creating 'Contributor' role assignment under scope '/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590'
  Retrying role assignment creation: 1/36
{
  ...
}
```